python/adelphi/tests/integration/__init__.py

```diff
@@ -62,6 +62,17 @@ def makeTempDirs(self):
         outputDir = os.path.join(base, "outputDir")
         os.mkdir(outputDir)
         self.dirs = TempDirs(base, outputDir)
+        self.addCleanup(self.cleanUpTempDirs)
+
+
+    def cleanUpTempDirs(self):
+        # TODO: Note that there's no easy way to access this from test-adelphi unless we modify the
+        # ini generation code... and I'm not completely sure that's worth it.  Might want to think
+        # about just deleting this outright... or making it a CLI option that can be easily accessed.
+        if "KEEP_LOGS" in os.environ:
+            log.info("KEEP_LOGS env var set, preserving logs/output at {}".format(self.dirs.basePath))
+        else:
+            shutil.rmtree(self.dirs.basePath)
```

Comment:
Switched to cleanup functions rather than an explicit tearDown() when trying to diagnose a different problem but in the end I decided to keep the change.  tearDown() does have some odd semantics; it's not called if setUp() doesn't succeed, for instance.  Rather than worry about it in the future it seemed saner to just convert to cleanup methods directly.